pallets/crowdloan-rewards/src/lib.rs

```diff
@@ -51,110 +71,54 @@ pub mod pallet {
 		#[pallet::weight(0)]
 		pub fn initialize_reward(
 			origin: OriginFor<T>,
-			contributions: Vec<(T::RelayChainAccountId, u32)>,
-			rate: u32,
+			contributions: Vec<(T::AccountId, BalanceOf<T>)>,
 			end_block: T::BlockNumber,
 		) -> DispatchResultWithPostInfo {
 			ensure_root(origin)?;
+
 			let now = frame_system::Pallet::<T>::block_number();
-			ensure!(
-				&now < &end_block,
-				Error::<T>::InvalidEndBlock
-			);
-			let current_reward_end_in = CurrentRewardEndIn::<T>::get();
+			ensure!(&now < &end_block, Error::<T>::InvalidEndBlock);
 
+			let current_reward_end_in = CurrentRewardEndIn::<T>::get();
 			ensure!(
 				&now >= &current_reward_end_in,
 				Error::<T>::AlreadyInitReward
 			);
-			for (account, amount) in &contributions {
-				let reward_info = RewardInfo {
-					total_reward: BalanceOf::<T>::from(*amount)
-						.saturating_mul(BalanceOf::<T>::from(rate)),
-					claimed_reward: 0u32.into(),
-					last_paid: now.clone(),
-				};
-				Contributors::<T>::insert(account, reward_info);
-			};
+
+			for (who, amount) in &contributions {
```

Comment:
we need to make sure the reward does not go over fund in pallet account. Otherwise reject it

Reply:
Added the validation of the reward
https://github.com/polkafoundry/newpolka/pull/55/commits/2ab33c221080d49511dd1ad89c9e64ffa03503e4#diff-2bbae40c9c7fdfeeb0b1708d368844303a595083bc948ccd684b577e5d13f6ebR112